### PR TITLE
Fix regression on datetime in MultiIndex

### LINKED ISSUE #35015: BUG: MultiIndex columns containing datetime cannot be used (regression from 1.0.5 to master)
- [x] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
In [1]: import pandas as pd

In [2]: pd.__version__
Out[2]: '1.1.0.dev0+1967.gf5b2e5a9a'

In [3]: expected_column_index = pd.MultiIndex.from_tuples(
   ...:     [(pd.to_datetime("02/29/2020"), pd.to_datetime("03/01/2020"))],
   ...:     names=[
   ...:         'a', 'b'])

In [4]: pd.DataFrame([], columns=expected_column_index)
---------------------------------------------------------------------------
InvalidIndexError
```
#### Problem description

This works in 1.0.5, but not in master.

See discussion in #34748 .  This new issue should be resolved before that PR moves forward.

#### Expected Output

```python
In [1]: import pandas as pd

In [2]: pd.__version__
Out[2]: '1.0.5'

In [3]:  expected_column_index = pd.MultiIndex.from_tuples(
   ...:        [(pd.to_datetime("02/29/2020"), pd.to_datetime("03/01/2020"))],
   ...:        names=[
   ...:            'a', 'b'])

In [4]: pd.DataFrame([], columns=expected_column_index)
Out[4]:
Empty DataFrame
Columns: [(2020-02-29 00:00:00, 2020-03-01 00:00:00)]
Index: []
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : f5b2e5a9a2431165b1e8002219c5d3962d7bc920
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.0.dev0+1967.gf5b2e5a9a
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 42.0.2.post20191201
Cython           : 0.29.20
pytest           : 5.4.3
hypothesis       : 5.18.0
sphinx           : 3.1.1
blosc            : None
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : 4.5.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.15.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.7.4
fastparquet      : 0.4.0
gcsfs            : None
matplotlib       : 3.1.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 0.15.1
pytables         : None
pyxlsb           : 1.0.6
s3fs             : 0.4.2
scipy            : 1.3.1
sqlalchemy       : 1.3.17
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.15.1
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.46.0
</details>

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] closes #35015
- [x] tests added / passed
  - `tests.indexing.multiindex.test_datetime:test_multiindex_datetime_columns`
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry
   - Not needed - this was a regression

### CODE REVIEW

**Comment by WillAyd** [comment]
@jbrockmendel this is what you had in mind right?

**Comment by jbrockmendel** [comment]
Yes

File: pandas/tests/indexing/multiindex/test_datetime.py [hunk 1]
```diff
@@ -20,3 +29,16 @@ def test_multiindex_period_datetime():
...
+
+def test_multiindex_datetime_columns():
+    # GH35015, using datetime as column indices raises exception
+
+    mi = MultiIndex.from_tuples(
+        [(to_datetime("02/29/2020"), to_datetime("03/01/2020"))], names=["a", "b"]
+    )
+
+    df = DataFrame([], columns=mi)
+    assert list(df.columns.names) == ["a", "b"]
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_datetime.py:42]: can you construct the expected dataframe and use assert_frame_equal?

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -1165,6 +1165,9 @@ def _convert_to_indexer(self, key, axis: int, is_setter: bool = False):
                     if len(key) == labels.nlevels:
                         return {"key": key}
                     raise
+            except InvalidIndexError:
```
Comment (jreback) [inline pandas/core/indexing.py:1168]: can you add a comment here about why this is

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew, bug fixes 1.1 in conversion section, patch looks good, some comments

**Comment by Dr-Irv** [comment]
> can you add a whatsnew, bug fixes 1.1 in conversion section, patch looks good, some comments

@jreback I'm not going to add a whatsnew, as this was NOT a bug in 1.0.5, but was introduced during the development of 1.1 .  I will do the other things.

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
> > can you add a whatsnew, bug fixes 1.1 in conversion section, patch looks good, some comments
> 
> @jreback I'm not going to add a whatsnew, as this was NOT a bug in 1.0.5, but was introduced during the development of 1.1 . I will do the other things.

thanks

**Review by jreback — APPROVED** [review]

**Pull request merged**